### ISSUE TITLE
[a11y]: remove all instances of -ms-high-contrast

### ISSUE THREAD

**Issue by ginabak** [Shopify/polaris-react#3826]
## Issue summary
There are many instances of `-ms-high-contrast`
See here: https://github.com/search?q=%22-ms-high-contrast%22+repo%3AShopify%2Fpolaris-react&type=code

According [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/-ms-high-contrast): 

> This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future

More on the limitations of `-ms-high-contrast`, [here](https://developer.paciellogroup.com/blog/2016/12/windows-high-contrast-mode-the-limited-utility-of-ms-high-contrast/).  

For even MORE context, see [this slack conversation](https://shopify.slack.com/archives/C03T74ZTD/p1607460676013100). 

## What should we do?  / Potential solutions

We must remove the media query since it only properly works in Edge and not recommended for production. 

For the use of an outline, this fix below is okay: 
From: 

```javascript
@media (-ms-high-contrast: active) {
    outline: 1px solid ms-high-contrast-color('text');
  }
```

to 

```javascript
outline: $border-width solid transparent;
```

However, in this [interesting article](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/), we know that we can use a media query ... 

```CSS
@media (forced-colors: active) {
  /* Styles here only apply to Forced Colors Mode */
}
```
that basically does the same thing as `-ms-high-contrast: active` **_HOWEVER_**, 😭 it does NOT work for firefox. 

In [this article ](https://hacks.mozilla.org/2020/07/adding-prefers-contrast-to-firefox/), it suggests creating a custom media query called `prefers-contrast`, which is quite laborious. 

We must then remove all `@media (forced-colors: active)` in the Polaris repo too. It is currently used in certain places as a temporary fix. 

There are still many instances of `-ms-high-contrast` in the Polaris repo. 
See here: https://github.com/search?q=%22-ms-high-contrast%22+repo%3AShopify%2Fpolaris-react&type=code

According [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/-ms-high-contrast):

> This feature is non-standard and is not on a standards track. Do not use it on production sites facing the Web: it will not work for every user. There may also be large incompatibilities between implementations and the behavior may change in the future

More on the limitations of `-ms-high-contrast`, [here](https://developer.paciellogroup.com/blog/2016/12/windows-high-contrast-mode-the-limited-utility-of-ms-high-contrast/).
